### DISCUSSION TITLE
paging-demo-upgraded-to-aspnet-mvc-beta/

### DISCUSSION THREAD

**Discussion by martijnboland** [martijnboland/blog#26] [Blog comments]
# paging-demo-upgraded-to-aspnet-mvc-beta/

https://blogs.taiga.nl/martijn/posts/paging-demo-upgraded-to-aspnet-mvc-beta/

**Comment by disqus-comments-migrator** [comment]
<em>zapper commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at januari 23 2009, 12:23</em>

---

<p>How can I open this in MS Visual Studio 2008 Express?  The lib folder cannot be accessed.  The reference to MvcPaging.dll is missing too.  Could you please post an Express version of the demo?  Thanks for these posts by the way!</p>

**Comment by disqus-comments-migrator** [comment]
<em>martijn commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at januari 23 2009, 07:07</em>

---

<p>Unfortunately, I don't have VS 2008 Express installed, so no luck I'm afraid. <br>Tried to install VWD Express but it seems to nicely integrate into my VS 2008 Pro environment so I can't get a clean VS 2008 Express environment.</p>

**Comment by disqus-comments-migrator** [comment]
<em>martijn commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at januari 23 2009, 07:10</em>

---

<p>Unfortunately I don't have a proper VS 2008 Express environment, so I can't create a solution.<br><br>But it isn't too hard to copy the source into a VS 2008 Express solution is it?</p>

**Comment by disqus-comments-migrator** [comment]
<em>Brian commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at januari 31 2009, 01:00</em>

---

<p>This doesn't seem to run in VS2008 with ASP.NET MVC RC1.</p>

**Comment by disqus-comments-migrator** [comment]
<em>martijn commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at januari 31 2009, 06:17</em>

---

<p>From RC, viewContext doesn't inherit from RequestContext anymore. in the Pager.cs GeneratePageLink method, you have to change the line<br><br>var virtualPathData = RouteTable.Routes.GetVirtualPath(this.viewContext, pageLinkValueDictionary);<br><br>to:<br><br>var virtualPathData = RouteTable.Routes.GetVirtualPath(this.viewContext.RequestContext, pageLinkValueDictionary);<br></p>

**Comment by disqus-comments-migrator** [comment]
<em>Brian commented [on Disqus](https://disqus.com/home/forum/martijnboland/) at februari 04 2009, 04:25</em>

---

<p>martijn:  Thanks, it works now.  Not sure why but I also had to point my references to the RC assemblies to get that new line of code to work.</p>
